code/modules/events/spacevine.dm

```diff
@@ -532,15 +535,22 @@
 		buckle_mob(V, 1)
 
 /obj/effect/spacevine/proc/spread()
-	var/direction = pick(cardinal)
-	var/turf/stepturf = get_step(src,direction)
-	for(var/datum/spacevine_mutation/SM in mutations)
-		SM.on_spread(src, stepturf)
-		stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen
-	if(!locate(/obj/effect/spacevine, stepturf))
-		if(stepturf.Enter(src))
-			if(master)
-				master.spawn_spacevine_piece(stepturf, src)
+	var/list/dir_list = cardinal.Copy()
+	while(dir_list.len)
+		var/direction = pick(dir_list)
+		dir_list -= direction
+		var/turf/stepturf = get_step(src,direction)
+		var/spread_success = FALSE
+		for(var/datum/spacevine_mutation/SM in mutations)
+			spread_success |= SM.on_spread(src, stepturf) // If this returns 1, spreading succeeded
+			stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen

```

Comment:
Yeah, the ref should point to the same datum - this is just a holdover from re-indenting existing code